### ISSUE TITLE
Feature Request: `AWS::Lambda::LayerVersion` - Verify `Content/S3Bucket`

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4429]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

Currently cfn-lint is unable to verify LambdaLayers `Content/S3Bucket`:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test


Resources:
  Layer:
    Type: AWS::Lambda::LayerVersion
    Properties:
      Content:
        S3Bucket: "arn:aws:s3:::01234567890-test"
        S3Key: "test.zip"
```

Error:

>1 validation error detected: Value 'arn:aws:s3:::01234567890-test' at 'content.s3Bucket' failed to satisfy constraint: Member must satisfy regular expression pattern: [0-9A-Za-z\.\-_]*(?<!\.)

### Describe the solution you'd like

A check which verifies if the value is correct and usable.

### Additional context

\-

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
